controllers/object_controls.go

```diff
@@ -773,6 +785,73 @@ func applyCommonDaemonsetConfig(obj *appsv1.DaemonSet, config *gpuv1.ClusterPoli
 	return nil
 }
 
+// apply necessary transforms if a custom host root path is configured
+func transformForHostRoot(obj *appsv1.DaemonSet, hostRoot string) {
+	if hostRoot == "" || hostRoot == "/" {
+		return
+	}
+
+	transformHostRootVolume(obj, hostRoot)
+	transformHostDevCharVolume(obj, hostRoot)
+}
+
+func transformHostRootVolume(obj *appsv1.DaemonSet, hostRoot string) {
+	containsHostRootVolume := false
+	for _, volume := range obj.Spec.Template.Spec.Volumes {
+		if volume.Name == "host-root" {
+			volume.HostPath.Path = hostRoot
+			containsHostRootVolume = true
+			break
+		}
+	}
+
+	if !containsHostRootVolume {
+		return
+	}
+
+	for index := range obj.Spec.Template.Spec.InitContainers {
```

Comment:
Reading this again, I can simplify by iterating over all containers and adding this envvar if the container has a volumeMount named "driver-root"